src/v/kafka/server/group.cc

```diff
@@ -2052,8 +2088,18 @@ group::store_txn_offsets(txn_offset_commit_request r) {
             co_await _partition->raft()->step_down(
               "group store_txn_offsets failed");
         }
-        co_return txn_offset_commit_response(
-          r, error_code::unknown_server_error);
+        auto tx_ec = map_tx_replication_error(result.error());
+        switch (tx_ec) {
+        case cluster::tx::errc::not_coordinator:
+            co_return txn_offset_commit_response(
+              r, error_code::not_coordinator);
+        case cluster::tx::errc::timeout:
+            co_return txn_offset_commit_response(
+              r, error_code::request_timed_out);
+        default:
+            co_return txn_offset_commit_response(
+              r, error_code::unknown_server_error);
+        }
```

Comment:
can we just use `kafka::map_tx_errc` here?

Reply:
yes, good point